crates/forge_domain/src/orch.rs

```diff
@@ -489,10 +489,12 @@ impl<A: Services> Orchestrator<A> {
             if empty_tool_calls {
                 // No tool calls present, which doesn't mean task is complete so reprompt the
                 // agent to ensure the task complete.
-                let content = self
-                    .services
-                    .template_service()
-                    .render("{{> partial-tool-required.hbs}}", &())?;
+                let content = self.services.template_service().render(
+                    "{{> partial-tool-required.hbs}}",
+                    &serde_json::json!({
+                        "stats": tool_context.stats.read().await.clone().unwrap_or_default(),
+                    }),
+                )?;
```

Comment:
- Above change isn't required
- On completion we should check the task list, and send the new task to the agent as a response (Should only happen in Act mode).
- In plan mode tasks will only be inserted